### PR TITLE
runtime-tester: fix runtime_fuzzer test

### PR DESCRIPTION
**Pull request by mina86**
This fixes the following error:

    error[E0308]: mismatched types
      --> fuzz_targets/runtime_fuzzer.rs:10:12
    10 |     if let Err(e) = runtime_stats {
       |            ^^^^^^   ------------- this expression has type `ScenarioResult<RuntimeStats, near_client_primitives::types::Error>`
       |            |
       |            expected struct `ScenarioResult`, found enum `Result`
    error[E0599]: no method named `unwrap` found for struct `ScenarioResult` in the current scope
      --> fuzz_targets/runtime_fuzzer.rs:14:35
    14 |         let stats = runtime_stats.unwrap();
       |                                   ^^^^^^ method not found in `ScenarioResult<RuntimeStats, near_client_primitives::types::Error>`

It can be reproduced by running runtime/fuzz.py pytest.

Issue: https://github.com/near/nearcore/issues/4835

### CODE REVIEW

**Comment by matklad** [comment]
cc @bowenwang1996, that's the example of fuzzing infra bitrotting I was talking about during the OKR meeting. 

Would be cool to add this to CI somehow (haven't resarched the best way to do this myself yet. )

**Review by matklad — APPROVED** [review]
LGTM, couple of stretch goals

* add some CI here
* if that's burdersome, move the body of `fuzz_target` into the library, so that its verified during cargo-check. I guess an elegant way to do that *would be* to put a functions instead of the closure inside:

```diff
diff --git a/test-utils/runtime-tester/fuzz/fuzz_targets/runtime_fuzzer.rs b/test-utils/runtime-tester/fuzz/fuzz_targets/runtime_fuzzer.rs
index a6b961fa9..a8a96316b 100644
--- a/test-utils/runtime-tester/fuzz/fuzz_targets/runtime_fuzzer.rs
+++ b/test-utils/runtime-tester/fuzz/fuzz_targets/runtime_fuzzer.rs
@@ -4,22 +4,4 @@ use runtime_tester::Scenario;
 use std::fs::File;
 use std::time::Duration;
 
-fuzz_target!(|scenario: Scenario| {
-    let filename = "failed_scenario.json";
-    let stats = match scenario.run().result {
-        Err(e) => {
-            serde_json::to_writer(&File::create(filename).unwrap(), &scenario).unwrap();
-            panic!("Bad scenario: {}, Error: {}", filename, e);
-        },
-        Ok(stats) => stats
-    };
-    for block_stats in stats.blocks_stats {
-        if block_stats.block_production_time > Duration::from_secs(1) {
-            serde_json::to_writer(&File::create(filename).unwrap(), &scenario).unwrap();
-            panic!(
-                "Bad scenario: {}, block at height {} was produced in {:?}",
-                filename, block_stats.height, block_stats.block_production_time
-            );
-        }
-    }
-});
+fuzz_target!(runtime_tester::Scenario::fuzzer::fuzz);
diff --git a/test-utils/runtime-tester/src/fuzzing.rs b/test-utils/runtime-tester/src/fuzzing.rs
index 21ad95202..5489e6e83 100644
--- a/test-utils/runtime-tester/src/fuzzing.rs
+++ b/test-utils/runtime-tester/src/fuzzing.rs
@@ -13,10 +13,10 @@ use nearcore::config::{NEAR_BASE, TESTING_INIT_BALANCE};
 use byteorder::{ByteOrder, LittleEndian};
 use libfuzzer_sys::arbitrary::{Arbitrary, Result, Unstructured};
 
-use std::collections::HashSet;
 use std::iter::FromIterator;
 use std::mem::size_of;
 use std::str::FromStr;
+use std::{collections::HashSet, fs::File, time::Duration};
 
 pub type ContractId = usize;
 
@@ -510,3 +510,23 @@ impl Function {
         (0, Some(20))
     }
 }
+
+pub fn fuzz(scenario: Scenario) {
+    let filename = "failed_scenario.json";
+    let stats = match scenario.run().result {
+        Err(e) => {
+            serde_json::to_writer(&File::create(filename).unwrap(), &scenario).unwrap();
+            panic!("Bad scenario: {}, Error: {}", filename, e);
+        }
+        Ok(stats) => stats,
+    };
+    for block_stats in stats.blocks_stats {
+        if block_stats.block_production_time > Duration::from_secs(1) {
+            serde_json::to_writer(&File::create(filename).unwrap(), &scenario).unwrap();
+            panic!(
+                "Bad scenario: {}, block at height {} was produced in {:?}",
+                filename, block_stats.height, block_stats.block_production_time
+            );
+        }
+    }
+}

```

alas, that' dosen't work fuzz-target sytnax wants closure: 

https://github.com/rust-fuzz/libfuzzer/blob/fcf3b18c854e2e5e7f61a8a701f72ef051b1c7e1/src/lib.rs#L138

**Comment by mina86** [comment]
> Would be cool to add this to CI somehow (haven't resarched the best way to do this myself yet. )

I’m working on adding alerting to NayDuck so those failures would be noticed.

File: test-utils/runtime-tester/fuzz/fuzz_targets/runtime_fuzzer.rs [hunk 1]
```diff
@@ -4,22 +4,23 @@ use runtime_tester::Scenario;
...
+    Ok(())
+}
+
+fn fuzz(scenario: Scenario) {
+    if let Err(err) = do_fuzz(&scenario) {
+        let file = "failed_scenario.json";
+        serde_json::to_writer(&File::create(file).unwrap(), &scenario).unwrap();
+        panic!("Bad scenario: {}, {}", file, err);
+    }
+}
```
Comment (matklad) [inline test-utils/runtime-tester/fuzz/fuzz_targets/runtime_fuzzer.rs:24]: Sorry, I was being confusing here. What I mean is extract the function *and* move the function into `runtime-tester/src/lib.rs`, such that it is type-checked when you do `cargo check --workspace` (fuzz tagets are not part of the workspace, and that's why we are getting a compilation error). *Just* extracting the function doesn't chang much (although I must say I find it more readable when mosht of the code is outsied of macro). 

Also, I've meant to approve this originally, this is a nice-to-have, but not blocking.

**Review by matklad — APPROVED** [review]

File: test-utils/runtime-tester/fuzz/fuzz_targets/runtime_fuzzer.rs (hunk 1 above)
Comment (mina86) [reply test-utils/runtime-tester/fuzz/fuzz_targets/runtime_fuzzer.rs:24]: OK, I’m just gonna merge it then since this is actively breaking NayDuck.

**Pull request merged**